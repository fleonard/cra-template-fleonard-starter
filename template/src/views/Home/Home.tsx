import React, { Fragment } from "react";
import { Container, Typography } from "@material-ui/core";

import Counter from "../../components/counter/Counter";

const Home: React.FC = () => {
  return (
    <Fragment>
      <Container maxWidth="lg">
      <Typography variant="h1">Create React App powered by fleonard</Typography>
      <Counter />
      </Container>
    </Fragment>
  );
};

export  default Home;
