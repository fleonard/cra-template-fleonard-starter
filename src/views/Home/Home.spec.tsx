import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import "jest-styled-components";

import configureStore from "redux-mock-store";

import Home from "./Home";

describe("About Component", () => {
  const mockStore = configureStore([]);
  const store = mockStore({
    count: {
      value: 42,
    },
  });

  it("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
