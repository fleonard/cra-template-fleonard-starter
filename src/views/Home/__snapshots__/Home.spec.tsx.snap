// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component renders correctly 1`] = `
.c3 {
  margin: 0 auto;
  padding-top: 40px;
}

.c0 {
  margin: 0 auto;
  padding-top: 40px;
}

.c1.c1 {
  font-size: 22px;
}

.c2.c2 {
  font-size: 16px;
}

.c4.c4 {
  margin-right: 20px;
}

<div
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <h1
    class="MuiTypography-root MuiTypography-h1"
  >
    Create React App powered by fleonard
  </h1>
  <div
    class="MuiGrid-root c0 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h2
      class="MuiTypography-root c1 card-title MuiTypography-h2"
    >
      Counter component
    </h2>
    <h4
      class="MuiTypography-root c2 MuiTypography-h4"
    >
      Counter: 
      <strong
        data-testid="counter"
      >
        42
      </strong>
    </h4>
    <div
      class="MuiGrid-root c3 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <button
        aria-label="decrement"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained c4 MuiButton-containedPrimary"
        data-testid="decrement"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
           
          -
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="increment"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained c4 MuiButton-containedPrimary"
        data-testid="increment"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
           
          +
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
